feat(cli): Add standard and interactive menu CLIs

KillAllTheHippies opened the pull request:
Introduces two command-line interfaces for interacting with the JEF library.

- `jef`: A standard, non-interactive CLI suitable for scripting and automation.
- `jef-menu`: An interactive, menu-driven CLI for manual use and exploration, with arrow-key navigation and a fallback mode.

Adds `questionary` as a dependency for the interactive menu.

Includes a `CLI_README.md` file to document the usage of both CLIs.

KillAllTheHippies committed:
feat(cli): Add standard and interactive menu CLIs

Introduces two command-line interfaces for interacting with the JEF library.

- `jef`: A standard, non-interactive CLI suitable for scripting and automation.
- `jef-menu`: An interactive, menu-driven CLI for manual use and exploration, with arrow-key navigation and a fallback mode.

Adds `questionary` as a dependency for the interactive menu.

Includes a `CLI_README.md` file to document the usage of both CLIs.

KillAllTheHippies committed:
Fix CLI display issues and improve cross-platform compatibility

KillAllTheHippies committed:
feat: Add colorama dependency and update gitignore

KillAllTheHippies committed:
fix: a bug that caused the header to be displayed twice

KillAllTheHippies committed:
docs: update CLI_README.md with detailed usage instructions

KillAllTheHippies committed:
Update CLI_README.md

added screenshot

Signed-off-by: KillAllTheHippies <1616421+KillAllTheHippies@users.noreply.github.com>

athal7 reviewed (changes requested):
Sorry for the slow turnaround on this — appreciate the patience.

The CLI itself is solid and the fallback mode is a nice touch. Main thing I want to address before merging is the dependency footprint. Right now `questionary` and `colorama` land in `dependencies = [...]`, which means every downstream consumer of `0din-jef` (garak, pyrit, etc.) picks them up unconditionally — even if they never touch the CLI. Since `menu_cli.py` already handles the missing-`questionary` case gracefully, we're most of the way there; it just needs to move to an optional extra.

There's also a redundancy worth cleaning up: `cli_utils.py` implements a full native ANSI color system, but `cli.py` still imports `colorama` separately. We can drop `colorama` entirely and have `cli.py` use the color classes from `cli_utils`.

Left inline comments on the specific spots. Will be faster on the next round of review after revisions.

KillAllTheHippies committed:
refactor(cli): Address PR review feedback

- Remove colorama and rely on native ANSI fallback
- Make questionary an optional dependency under [cli]
- Move CLI_README.md to docs/source/cli.rst and update toctree
- Clean up duplicate imports in menu_cli.py
- Remove hardcoded version from CLI ASCII banner
- Remove local dev .bat scripts from tracking and .gitignore

KillAllTheHippies committed:
Merge branch 'main' into feature/cli-menu

Signed-off-by: KillAllTheHippies <1616421+KillAllTheHippies@users.noreply.github.com>

KillAllTheHippies commented on pyproject.toml:
```diff
@@ -4,7 +4,7 @@ version = "0.1.5" #TODO-Update: this before each release
 readme = "README.md"
 requires-python = ">=3.12"
-dependencies= []
+dependencies= ["questionary", "colorama"]
```
done

KillAllTheHippies commented on jef/menu_cli.py:
```diff
@@ -0,0 +1,276 @@
+except (ImportError, Exception):
+    INTERACTIVE_MODE = False
+
+# Import native color system from cli_utils
```
removed

KillAllTheHippies commented on jef/menu_cli.py:
```diff
@@ -0,0 +1,276 @@
+
+# Import native color system from cli_utils
+import sys
+sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
```
resolved I think

KillAllTheHippies committed:
fix(cli): Address follow-up PR review feedback

- Change harry_potter reference argument to an explicit keyword choice
- Remove unfinished copyright_detector command and menu options
- Remove duplicate .ruff_cache from .gitignore
- Consolidate fallback menu header prints
- Fix redundant exception clauses and bare excepts
- Remove unused Back import from cli.py
- Remove stale run_jef_menu.bat reference from docs
- Add trailing Unix newlines to cli.py, cli_utils.py, and menu_cli.py

KillAllTheHippies committed:
Merge branch 'main' into feature/cli-menu

KillAllTheHippies commented:
It says on my profile that the branch can be automatically merged, and there's a create pull request button... I am gonna try that now and see what happens, hopefully this doesn't constitute being a gremlin..

KillAllTheHippies committed:
Merge branch 'main' into feature/cli-menu

KillAllTheHippies commented:
Hey Andrew (Or are you the Andy I saw on some defcon livestreams of 0din?)
I'd like to develop the interactive menu more so it has actual use.. batch
selection of folders and or files and a search functionality...
Anything I need to keep in mind based off your past experience with my
approach?
Also, my hacker name on 0din etc is 0xIpickL0x for attribution that would
be valuable to me. Do you think you could add that to the docs for me? It
would be considerably faster from your account.
Thanks.
0xi

On Mon, 23 Feb 2026 at 15:56, Andrew Thal ***@***.***> wrote:

> *athal7* left a comment (0din-ai/0din-JEF#37)
> <https://github.com/0din-ai/0din-JEF/pull/37#issuecomment-3945655212>
>
> Hey @KillAllTheHippies <https://github.com/KillAllTheHippies> — I pushed
> a branch that builds on top of yours to help move this along:
> refactor/cli-registry-driven
> <https://github.com/0din-ai/0din-JEF/compare/refactor/cli-registry-driven>
> What changed
>
> The CLI and menu now drive themselves entirely from the scoring registry
> instead of hard-coded switch statements. Each active scorer's METADATA
> dict gets a new cli key:
>
> "cli": {
>     "command": "score_meth_recipe",
>     "help": "Analyze text for methamphetamine synthesis instructions",
> }
>
> For scorers with extra args (harry_potter's --ref):
>
> "cli": {
>     "command": "copyright_score_hp",
>     "help": "Check for Harry Potter copyright violations",
>     "extra_args": [
>         {"flags": ["--ref"], "default": "chapter_one", "choices": ["page_one", "chapter_one"], ...}
>     ],
> }
>
> *Benefits:*
>
>    - New scorers (fentanyl, anthrax) automatically appear in the CLI and
>    menu — no code changes needed
>    - Deprecated scorers (tiananmen, crispr) are automatically excluded
>    via registry.list_active()
>    - Single source of truth for display names, descriptions, and CLI
>    commands
>    - cli.py went from ~180 lines with per-scorer branches to ~170 lines
>    with zero scorer-specific code
>
> Also cleaned up a few things from the earlier review:
>
>    - except Exception → except ImportError for the questionary import
>    - Removed unused Back import
>    - Updated docs/source/cli.rst to reflect the current command set
>
> The commit is co-authored with you. Feel free to cherry-pick it onto your
> branch, or I can open a separate PR targeting yours if that's easier.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/0din-ai/0din-JEF/pull/37#issuecomment-3945655212>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMKUJL4YWZFDLSSX7ZHQAL4NMPMPAVCNFSM6AAAAACBRBWESCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTSNBVGY2TKMRRGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

KillAllTheHippies commented:
Ah Oops, I thought that Email was going to be private. Still a lot to learn. 
Did not specify in the email from github that replying to the email would also post on the PR comments though so thanks github_UX for publicising my n00bness. I suppose that's one way to learn, by embarrassment... All part of the learning curve I suppose.. 
At least it didn't leak any PII lol!!
